DEPRECATIONS.rst

```diff
@@ -0,0 +1,101 @@
+Deprecations: how to migrate
+============================
+
+New Tracer interface
+--------------------
+
+The following classes and methods are deprecated and should not be used.
+Please use the Thread interface instead.
+They will be removed in version 1.0.
+
+REASON: they don't work well in multi-threaded environments and can cause
```

Comment:
This is really excellent. To have reasoning behind deprecations is something that i think end users will appreciate